Vwitter/View Controllers/LoginViewController.m

```diff
@@ -20,27 +19,8 @@ @implementation LoginViewController
 - (void)viewDidLoad {
     [super viewDidLoad];
     // Do any additional setup after loading the view.
-}
-
-- (void)registerUser {
-    // initialize a user object
-    PFUser *newUser = [PFUser user];
-    
-    // set user properties
-    newUser.username = self.usernameField.text;
-    newUser.password = self.passwordField.text;
-    
-    // call sign up function on the object
-    [newUser signUpInBackgroundWithBlock:^(BOOL succeeded, NSError * error) {
-        if (error != nil) {
-            NSLog(@"Error: %@", error.localizedDescription);
-        } else {
-            NSLog(@"User registered successfully");
-            [self performSegueWithIdentifier:@"loginSegue" sender:nil];
-            
-            // manually segue to logged in view
-        }
-    }];
+    self.usernameField.delegate = self;
+    self.passwordField.delegate = self;
 }
 
 - (void)loginUser {
```

Comment:
nit: in this `loginUser` method, you are directly referring to `self` inside of the `logInWithUsernameInBackground:username` completion block. Make sure to do the weakSelf/strongSelf pattern to avoid keeping things in memory long than they need to be

Reply:
Ah yup, my bad; hopefully this is the last time you have to remind me about that